### PR TITLE
Hive metastore

### PR DESCRIPTION
**Pull request by Ishiihara**
@ewencp The Hive metastore service

### CODE REVIEW

File: muckrake/services/hive.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            "collect_default": True
+        }
+    }
+
+    def __init__(self, context, mysql, hadoop):
+        super(HiveMetaStore, self).__init__(context, 1)
+        self.mysql = mysql
+        self.mysql_client = MysqlClient(context, self.mysql.node, self.mysql.connect_host())
+        self.hadoop = hadoop
+        self.hive_dir = "/opt/hive"
```
Comment (ewencp) [inline muckrake/services/hive.py]: duplicate of `HIVE_HOME`

File: muckrake/services/hive_client.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import re
+
+
+class HiveClient():
```
Comment (ewencp) [inline muckrake/services/hive_client.py]: should inherit from `object` so you get a new-style Python class

File: muckrake/services/hive_client.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            if col_type in self.int_types:
+                self.convert_funcs[col_type] = int
+            elif col_type in self.float_types:
+                self.convert_funcs[col_type] = float
+            elif col_type in self.datetime_types:
+                self.convert_funcs[col_type] = lambda s: s
+            else:
+                raise Exception("%s type is not yet supported." % col_type)
+
+    def create_database(self, database):
```
Comment (ewencp) [inline muckrake/services/hive_client.py]: How much of this code is basically the same as `MysqlClient`? Could we define a common base class, in most cases override only `execute_ddl` and `execute_query` to be database-specific, and then override the occasional additional method where there are other slight differences?

**Comment by ewencp** [comment]
Had some comments about minor cleanup, but otherwise LGTM (and checked that the new test passes cleanly).

File: muckrake/services/hive_client.py (hunk 3 above)
Comment (Ishiihara) [inline muckrake/services/hive_client.py]: Will do that after release. :)

**Pull request merged**